website/content/en/preview/upgrading/v1-migration.md

```diff
@@ -184,11 +184,65 @@ then you have to manually add more EC2NodeClasses and point their NodePools to t
+Karpenter will crash on the upgrade if NodePool resources contain this annotation. Users should make sure that their kubelet configurations are migrated prior to upgrading to properly drift their nodes to the correct configurations.
 
 ### Downgrading
-// TODO 
+The patch version of the v1beta1 Karpenter controller that include the conversion wehooks: 
+
+* v0.37.1
+* v0.36.3
+* v0.35.6
+* v0.34.7
+* v0.33.6
+
+{{% alert title="Note" color="warning" %}}
+On Rollback users can potentially experience data loss for values that were configured using v1 APIs. The conversion webhooks will convert configuration. If you are relying on kubeletConfiguration, and has been removed `compatibility.karpenter.sh/v1beta1-kubelet-conversion` annotation, rollback is not supported without replacing your nodes between EC2NodeClass and NodePool.
+{{% /alert %}}
+
+{{% alert title="Note" color="warning" %}}
+As both v1beta1 and v1 will be served, `kubectl` will require specifying v1beta1 version of the CRDs. For example: `k get nodeclaim.v1beta1.karpenter.sh`
+{{% /alert %}}
+
+Step 1.) Rollback the Karpenter Policy to reflect the version
+
+```bash
+TEMPOUT=$(mktemp)
+curl -fsSL https://raw.githubusercontent.com/aws/karpenter-provider-aws/v"${KARPENTER_VERSION}"/website/content/en/docs/getting-started/getting-started-with-karpenter/cloudformation.yaml > ${TEMPOUT} \
+    && aws cloudformation deploy \
+    --stack-name "Karpenter-${CLUSTER_NAME}" \
+    --template-file "${TEMPOUT}" \
+    --capabilities CAPABILITY_NAMED_IAM \
+    --parameter-overrides "ClusterName=${CLUSTER_NAME}"
+```
+
+Step 2.) Rollback the CRDs to the version that is being rollbacked
+
+```bash
+KARPENTER_NAMESPACE=kube-system 
+helm upgrade --install karpenter-crd oci://public.ecr.aws/karpenter/karpenter-crd --version "1.0.0" --namespace "${KARPENTER_NAMESPACE}" --create-namespace \
+  --set webhook.enabled=true \
+  --set webhook.serviceName=karpenter \
+  --set webhook.serviceNamespace="${KARPENTER_NAMESPACE}" \
+  --set webhook.port=8443
+```
+
+Step 3.) Rollback the Karpenter Controller Version 
+```bash
+helm upgrade --install karpenter oci://public.ecr.aws/karpenter/karpenter --version ${KARPENTER_VERSION} --namespace kube-system --create-namespace \
+  --set serviceAccount.annotations."eks\.amazonaws\.com/role-arn"=${KARPENTER_IAM_ROLE_ARN} \
+  --set settings.aws.defaultInstanceProfile=KarpenterNodeInstanceProfile-${CLUSTER_NAME} \
+  --set settings.clusterName=${CLUSTER_NAME} \
+  --set settings.interruptionQueue=${CLUSTER_NAME} \
+  --set controller.resources.requests.cpu=1 \
+  --set controller.resources.requests.memory=1Gi \
+  --set controller.resources.limits.cpu=1 \
+  --set controller.resources.limits.memory=1Gi \
+  --set webhook.enabled=true \
```

Comment:
Does this have any interactions with the `validatingwebhookconfigurations` and `mutatingwebhookconfigurations` that would be included when the webhook was enabled on the lower versions?

Reply:
I have not seen any interaction during my testing. The main thing would the logs they would see for their validation webhooks

Reply:
We took the position that both validating/mutating admission webhooks  will be running along side the conversion webhook, so users can't disable one without the other

Reply:
Fair, I think we should still say something like "make sure you're on v1.25+ before starting your migration